modules/openapi-generator/src/test/resources/3_0/oneOf_nonPrimitive.yaml

```diff
@@ -0,0 +1,29 @@
+openapi: 3.0.1
+info:
+  version: 1.0.0
+  title: Example - oneOf data type
+  license:
+    name: MIT
+servers:
+  - url: http://api.example.xyz/v1
+paths:
+  /example:
+    get:
+      operationId: list
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Example"
+components:
+  schemas:
+    Example:
+      oneOf:
+      - type: string
+        format: uuid
+      - type: string
+        format: date-time
+      - type: integer
+      - type: number
```

Comment:
@wing328 Opps, I didn't try java client generator with library: `okhttp-gson`, the result failed to compile because cannot find symbol method `validateJsonObject(JsonObject)` :(
I tested  java client generator with library: `native` & `jersey2`, th both generated clients passed compile.

Reply:
@wing328 thanks a lot